src/components/Stepper/Stepper.tsx

```diff
@@ -164,99 +200,86 @@ const Stepper = (props: StepperProps) => {
                 data-testid={`stepper-${index}`}
                 key={index}
               >
-                <div className="nitrozen-flex-center bar-ball-container">
-                  {/* Horizontal Stepper Circle states */}
-                  {isHorizontal && isActive(index) && (
-                    <div className="nitrozen-circle-outer-container">
-                      <SvgCircleProgress className="progress-svg" />
-                    </div>
-                  )}
-                  {isHorizontal && isChecked(index) && (
-                    <div className="nitrozen-cirle-check-container">
-                      <SvgCheckCircle className="completed-svg" />
-                    </div>
-                  )}
-                  {isHorizontal && isDisabled(index) && (
-                    <div className="nitrozen-circle-outer-container">
-                      <SvgCircleDisabled />
-                    </div>
-                  )}
-
-                  {/* Vertical Stepper Circle states */}
-                  {isVertical && item.isCompleted && (
-                    <div className="nitrozen-cirle-check-container">
-                      <div className="nitrozen-circle-outer"></div>
-                      <div className="nitrozen-checkmark"></div>
+                {/* Horizontal Stepper Circle states */}
+                {isHorizontal && (
+                  <div className="nitrozen-circle-outer-container">
```

Comment:
The vertical and horizontal component could be separated as a separate component.
Separating them would have helped to align with the single responsibility principle.
Would have also increasing the performance by avoiding re-render multiple elements.
The recommendation would to follow atomic design methodology.